x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/annotations/Annotation.java

```diff
@@ -14,15 +14,44 @@
 import org.elasticsearch.common.xcontent.ObjectParser;
 import org.elasticsearch.common.xcontent.ToXContentObject;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
 import org.elasticsearch.xpack.core.common.time.TimeUtils;
 import org.elasticsearch.xpack.core.ml.job.config.Job;
 
 import java.io.IOException;
 import java.util.Date;
+import java.util.Locale;
 import java.util.Objects;
 
 public class Annotation implements ToXContentObject, Writeable {
 
+    public enum Type {
+        ANNOTATION;
+
+        public static Type fromString(String value) {
+            return valueOf(value.toUpperCase(Locale.ROOT));
+        }
+
+        @Override
+        public String toString() {
+            return name().toLowerCase(Locale.ROOT);
+        }
+    }
+
+    public enum Event {
+        DELAYED_DATA,
+        MODEL_SNAPSHOT_STORED;
```

Comment:
We need to think about what the UI will put in this field for user-added annotations.  `user`?  `manual`?  Or leave it `null`.  Please can you talk to the UI team about this.

Reply:
FYI: The UI team settled on `user`. I've added `USER` enum value.